x-pack/filebeat/input/gcppubsub/otel_test.go

```diff
@@ -175,6 +175,9 @@ processors:
 		"agent.ephemeral_id",
 		"agent.id",
 		"event.created",
+		// only present in beats receivers
+		"agent.otelcol.component.id",
+		"agent.otelcol.component.kind",
```

Comment:
Any specific reason why we have the otel test for this specific input ?

Reply:
We have this task that aims to cover all non-metrics input to prove they work the same way inside a beats receiver and standalone beats https://github.com/elastic/ingest-dev/issues/4889. Currently, we only have OTEL integration tests for `gcppubsub`, `httpjson`, and `filestream`, so all tests were updated to include the new fields. Only `gcppubsub` has its own test file, while `filestream` and `httpjson` are covered in otel_test.go inside x-pack/filebeat.

Reply:
We have plans to cover more inputs in the task I linked, which will be addressed at a later time. The idea is to keep the OTel integration test closer to the input code from now on (started with gcppubsub). This task is only about adding the new fields and adjusting the tests to account for them.